mllib/src/main/scala/org/apache/spark/mllib/clustering/PowerIterationClustering.scala

```diff
@@ -337,7 +343,10 @@ object PowerIterationClustering extends Logging {
   def initDegreeVector(g: Graph[Double, Double]): Graph[Double, Double] = {
     val sum = g.vertices.values.sum()
     val v0 = g.vertices.mapValues(_ / sum)
-    Graph(VertexRDD(v0), g.edges)
+    val graph = Graph(VertexRDD(v0), g.edges)
+    materialize(graph)
+    g.unpersist()
```

Comment:
Unpersisting the input Graph at `randomInit` and `initDegreeVector` is actually unpersisting the same Graph. I think it is better to unpersist it in `run`:

```scala
def run(graph: Graph[Double, Double]): PowerIterationClusteringModel = {
    val w = normalize(graph)
    val w0 = initMode match {
      case "random" => randomInit(w)
      case "degree" => initDegreeVector(w)
    }
    w.unpersist()
    pic(w0)
  }
```

It looks more reasonable, as it looks a bit strange that a function unpersists an input from outside.

Reply:
Updated. Thanks